strider/trapi.py

```diff
@@ -390,3 +391,112 @@ def build_unique_kg_edge_ids(message: Message):
                 for eb in edge_binding_list:
                     if eb["id"] == edge_id:
                         eb["id"] = new_edge_id
+
+
+def is_valid_node_binding(message, nb, qgraph_node):
+    """
+    Check whether the given kgraph node
+    satisifies the given qgraph node
+    """
+    if qgraph_node.get("id", None) is not None:
+        if nb["id"] not in qgraph_node["id"]:
+            return False
+
+    kgraph_node = message["knowledge_graph"]["nodes"][nb["id"]]
+    if qgraph_node.get("category", None) is not None:
+        # Build list of allowable categories for kgraph nodes
+        qgraph_allowable_categories = []
+        for c in ensure_list(qgraph_node["category"]):
+            qgraph_allowable_categories.extend(
+                WBMT.get_descendants(c)
```

Comment:
Don't we have the "augmented" query graph where this is already done? Does it make sense to validate against that instead?

Reply:
The augmentation is currently done on the operation graph inside the query planner. Also, we can't use the `add_descendants` method because it only works on query graphs where predicates are lists. I could probably turn the add_descendants method into two methods (add_category_descendants and add_predicate_descendants) but I think at that point it's probably more complex than the current setup. If you have any thoughts on how to lay this out better let me know.